src/Interpreters/HashJoin.h

```diff
@@ -351,7 +354,15 @@ class HashJoin : public IJoin
     void reuseJoinedData(const HashJoin & join);
 
     RightTableDataPtr getJoinedData() const { return data; }
-    BlocksList releaseJoinedBlocks();
+    BlocksList releaseJoinedBlocks(bool restructure = false);
+
+    Block structureRightBlock(const Block & block) const;
```

Comment:
We manipulate with some 'structured' block and 'sample' block, - it'd be nice to clarify the purpose of them and difference (in comments)

Reply:
`structureRightBlock` is not used anymore, should delete it.
Updated comment for the replacement `prepareRightBlock`